### PR TITLE
[4.13] OCPBU-142 Add support to Confidential Computing on GCP

### PR DESCRIPTION
**Pull request by jianli-wei**
- The 4.13 feature [OCPBU-142 Add support to Confidential Computing on GCP](https://issues.redhat.com/browse/OCPBU-142)
- add workflow `cucushift-installer-rehearse-gcp-ipi-confidential-computing` and the provision chain
- add step `ipi-conf-gcp-confidential-computing` which inserts confidential computing settings into install-config.yaml
- add step `ipi-conf-gcp-control-plane-node-type` to set machine type for control-plane nodes (FYI https://cloud.google.com/compute/confidential-vm/docs/os-and-machine-type), btw the existing step `ipi-conf-gcp` doesn't support setting a specific machine type for control-plane nodes although it supports it for compute nodes
- add periodic jobs of `gcp-ipi-confidential-computing-p2-f14` and `gcp-ipi-confidential-computing-p2-f28-destructive`
- FYI the installation failure is due to [OCPBUGS-7582 RHCOS misses udev rules for GCE PD NVMe disks](https://issues.redhat.com/browse/OCPBUGS-7582)

### CODE REVIEW

**Comment by jianli-wei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehearse-gcp

**Comment by jianli-wei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehearse-gcp

**Comment by jianli-wei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehearse-gcp

**Comment by jianli-wei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehearse-gcp

**Comment by jianli-wei** [comment]
@jianlinliu @gpei please help review, thanks!

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml [hunk 1]
```diff
@@ -1809,6 +1809,26 @@ tests:
     test:
     - chain: openshift-e2e-test-qe-destructive
     workflow: cucushift-installer-rehearse-gcp-ipi-compact
+- as: gcp-ipi-confidential-computing-p2-f7
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml:1812]: I guess the feature does not affect the cluster's integration testing too much, I guess we can decrease the test frequency  a bit to f14.

**Comment by jianlinliu** [comment]
cc @liangxia for approve

**Comment by liangxia** [comment]
The required job `ci/prow/generated-config` for merge failed.

**Comment by jianlinliu** [comment]
@jianli-wei "generated-config" job failed.

**Comment by jianli-wei** [comment]
> The required job `ci/prow/generated-config` for merge failed.

Resolved, thanks!

**Pull request merged**